snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedConceptCreateRequest.java

```diff
@@ -140,9 +140,10 @@ private Concept convertConcept(final TransactionContext context) {
 	}
 
 	private void convertDescriptions(TransactionContext context, final String conceptId) {
+		TransactionContext newContext = context.inject().bind(Synonyms.class, new Synonyms(context)).build();
```

Comment:
Domain-specific requests should not alter the incoming context, they should only use the services provided by them. `SnomedConceptCreateRequest` instances are usually being executed inside a `com.b2international.snowowl.snomed.datastore.request.SnomedBulkRequest<R>`, which already binds the `Synonyms` service to the context.
Executing `SnomedConceptCreateRequest` instances outside of `SnomedBulkRequest` is allowed, but in those cases, the context should be able to provide the `Synonyms` cache/service, meaning that the creator of the context should inject the necessary services before passing them to the requests.